pytorchvideo/neural_engine/retrieval_hook.py

```diff
@@ -0,0 +1,104 @@
+import cv2
+import torch
+
+
+def process_video(video_path, model, max_frames):
```

Comment:
This is a file in PTV core lib folder so we will not create "pipeline" code into this folder, could we move to use (and possibly modify) with the decodeHook we talked about?

```
 
class DecodeHook(HookBase):
 def __init__(
   self,
   executor: Callable = full_decode,
   decode_audio: bool = True,
   decoder: str = DecoderType.PYAV,
   fail_strategy = "RAISE_ERROR",
   priority = 0,
 ):
   # Decoding params
   self.decode_audio = decode_audio
   self.decoder = decoder
   # Hook params
   self.executor = executor
   self.inputs = ["path"]
   self.outputs = ["video", "audio"] if decode_audio else ["video"]  # C T H W, for D2 T x( C H W )
   self.fail_strategy = fail_strategy
   self.priority = priority
 
 def _run(self, status: OrderedDict, ):
   frames = self.executor(status, decode_audio=self.decode_audio, decoder=self.decoder)
   return frames
 
 
def temporal_sample(frames_inds, frame_size, interval):
 def group_elements(n, iterable, padvalue='x'):
     return itertools.zip_longest(*[iter(iterable)]*n, fillvalue=padvalue)
 for output in group_elements(frame_size * interval, frames, padvalue=frames_inds[-1]):
     inds = list(itertools.islice(output, 0, len(output), interval))
     return frames[inds]  # frames[inds][:, 0, :, :]
 
 
class TemporalSampleHook(HookBase):
 def __init__(
   self,
   executor: Callable = full_decode,
   decode_audio: bool = True,
   fail_strategy = "RAISE_ERROR",
   priority = 0,
 ):
   # Decoding params
   self.decode_audio = decode_audio
   self.decoder = decoder
   # Hook params
   self.executor = executor
   self.inputs = ["path"]
   self.outputs = ["video", "audio"] if decode_audio else ["video"]
   self.fail_strategy = fail_strategy
   self.priority = priority
 
 def _run(self, status: OrderedDict, ):
   frames = self.executor(status, decode_audio=self.decode_audio, decoder=self.decoder)
   return frames
 
```